pkg/graveler/ref/manager.go

```diff
@@ -47,23 +47,75 @@ func NewKVRefManager(executor batch.Batcher, kvStore kv.StoreMessage, db db.Data
 }
 
 func (m *KVManager) GetRepository(ctx context.Context, repositoryID graveler.RepositoryID) (*graveler.Repository, error) {
-	return m.db.GetRepository(ctx, repositoryID)
+	repo := graveler.RepositoryData{}
+	_, err := m.kvStore.GetMsg(ctx, graveler.RepoPartition(), []byte(graveler.RepoPath(repositoryID)), &repo)
+	if err != nil {
+		if errors.Is(err, kv.ErrNotFound) {
+			err = graveler.ErrRepositoryNotFound
+		}
+		return nil, err
+	}
+	return graveler.RepoFromProto(&repo).Repository, nil
+}
+
+func (m *KVManager) createBareRepository(ctx context.Context, repositoryID graveler.RepositoryID, repository graveler.Repository) error {
+	repoRecord := &graveler.RepositoryRecord{
+		RepositoryID: repositoryID,
+		Repository:   &repository,
+	}
+	repo := graveler.ProtoFromRepo(repoRecord)
+	err := m.kvStore.SetMsgIf(ctx, graveler.RepoPartition(), []byte(graveler.RepoPath(repositoryID)), repo, nil)
+	if err != nil {
+		if errors.Is(err, kv.ErrPredicateFailed) {
+			err = graveler.ErrNotUnique
+		}
+		return err
+	}
+	return nil
 }
 
 func (m *KVManager) CreateRepository(ctx context.Context, repositoryID graveler.RepositoryID, repository graveler.Repository, token graveler.StagingToken) error {
-	return m.db.CreateRepository(ctx, repositoryID, repository, token)
+	firstCommit := graveler.NewCommit()
+	firstCommit.Message = graveler.FirstCommitMsg
+	firstCommit.Generation = 1
+	// commitID := m.addressProvider.ContentAddress(firstCommit)
+
+	err := m.createBareRepository(ctx, repositoryID, repository)
+	if err != nil {
+		return err
+	}
+
+	commitID, err := m.AddCommit(ctx, repositoryID, firstCommit)
+	if err != nil {
+		return err
+	}
+
+	err = m.CreateBranch(ctx, repositoryID, repository.DefaultBranchID, graveler.Branch{CommitID: commitID, StagingToken: token})
+	if err != nil {
+		return err
```

Comment:
Try to delete commit if create branch failed?

Reply:
How about the repo? Try to delete that too?
How do we handle failure in the delete?
What about failures such as crash after repo is created?

Reply:
only the repo was created ~= bare repo => legal state, the user can then attempt to create the default branch manually
repo and commit created but no branch => dangling commit

If we fail the cleanup - then I guess we don't really have anything else to do...